src/Common/MessageHandler.ts

```diff
@@ -12,6 +15,34 @@ export interface CachedDataPromise<T> {
 
 export const RequestMap: Record<string, CachedDataPromise<any>> = {};
 
+export function addExplorerMessageHandlers(explorer: Explorer) {
+  window.addEventListener(
+    "message",
+    (event) => {
+      if (isInvalidParentFrameOrigin(event)) {
+        return;
+      }
+
+      if (!shouldProcessMessage(event)) {
+        return;
+      }
+      explorer = new Explorer();
```

Comment:
Why do we need to create another explorer object here? Why not use the explorer that's passed through the parameter?